### PR TITLE
Fix crash when encountering a parenthesized expression when converting an 'if' to a 'switch'

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/21360

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -120,8 +120,10 @@ protected override IPattern<CasePatternSwitchLabelSyntax> CreatePatternFromExpre
                             // separate out the leftmost expression and reconstruct the rest of conditions into a new
                             // expression, so that they would be evaluated entirely after "expr" e.g. "expr && (cond1 && cond2)".
                             // Afterwards, we can directly use it in a case guard e.g. "case pat when cond1 && cond2:".
-                            var condition = (leftmost.Parent.Parent as BinaryExpressionSyntax)
-                                ?.WithLeft(((BinaryExpressionSyntax)leftmost.Parent).Right);
+                            var parentBinary = leftmost.GetAncestor<BinaryExpressionSyntax>();
```
Comment (sharwell) [inline src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs]: ❔ The use of `GetAncestor` makes me uncomfortable (things like nested blocks/lambdas are difficult to reason about). Why not `WalkUpParentheses`?

File: src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.cs [hunk 2]
```diff
@@ -875,5 +875,453 @@ int M(int i)
...
+            return;
+    }
+}",
+@"class C
+{
+    void M(int i)
+    {
+        switch (i)
+        {
+            case 1 when (i == 2) && i == 3: return;
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.cs]: ❔ Do we already have a bug filed for this? The `return;` wasn't on the same line before, so it shouldn't be here either.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.cs]: These tests aren't testing formatting (they impplicitly pass ignoreTrivia: true).  So the formatting isn't something that is actually being asserted.

File: src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs]: That's fair.  However, i thought this was a bit clearer while also being correct.  But i do see your point.   There is a nice symettry about WalkUpParentheses matching the WalkDownParentheses in GetLeftmostCondition.  I'll consider it :)

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest please

**Pull request merged**